crates/utils/src/error.rs

```diff
@@ -97,10 +117,16 @@ impl Display for LemmyError {
 
 impl actix_web::error::ResponseError for LemmyError {
   fn status_code(&self) -> http::StatusCode {
-    match self.inner.downcast_ref::<diesel::result::Error>() {
-      Some(diesel::result::Error::NotFound) => http::StatusCode::NOT_FOUND,
-      _ => http::StatusCode::BAD_REQUEST,
+    if let Some(diesel::result::Error::NotFound) =
+      self.inner.downcast_ref::<diesel::result::Error>()
+    {
+      return http::StatusCode::NOT_FOUND;
     }
+    if let Some(HttpUnauthorizedError) = self.inner.downcast_ref::<HttpUnauthorizedError>() {
+      return http::StatusCode::UNAUTHORIZED;
+    }
+
+    http::StatusCode::BAD_REQUEST
```

Comment:
Bit awkward to to HTTP status codes like this but I cant really think of a better solution.